src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -7936,9 +7936,34 @@ private SwitchStatementSyntax ParseSwitchStatement()
         {
             Debug.Assert(this.CurrentToken.Kind == SyntaxKind.SwitchKeyword);
             var @switch = this.EatToken(SyntaxKind.SwitchKeyword);
-            var openParen = this.EatToken(SyntaxKind.OpenParenToken);
             var expression = this.ParseExpressionCore();
-            var closeParen = this.EatToken(SyntaxKind.CloseParenToken);
+            SyntaxToken openParen;
+            SyntaxToken closeParen;
+            if (expression.Kind == SyntaxKind.ParenthesizedExpression)
+            {
+                var parenExpression = (ParenthesizedExpressionSyntax)expression;
+                openParen = parenExpression.OpenParenToken;
+                expression = parenExpression.Expression;
+                closeParen = parenExpression.CloseParenToken;
+
+                // PROTOTYPE(patterns2): Do we need to preserve any diagnostics that were attached directly to parenExpression?
+                Debug.Assert(parenExpression.GetDiagnostics().Length == 0);
+            }
+            else if (expression.Kind == SyntaxKind.TupleExpression)
+            {
+                // As a special case, when a tuple literal is the governing expression of
+                // a switch statement we permit the switch statement's own parentheses to be omitted.
+                // LDM 2018-04-04.
+                openParen = closeParen = default;
+            }
```

Comment:
Can you add an example of the newly supported syntax to the "all syntax" test? I don't remember the exact name for that resource, but I can dig it up if needed.

Reply:
Since the new syntax forms are not tied to any language version yet, not yet.

---
In reply to: [179894724](https://github.com/dotnet/roslyn/pull/25963#discussion_r179894724) [](ancestors = 179894724)